### PR TITLE
AN-109087: Add time/date decoder in kdbgo

### PR DESCRIPTION
**Pull request by yznima**
* Cleanup the decode code
* Removed use of named parameters
* Added decoder for scalar date
* Moved most of decode's test to encode testcases to increase covergae for both
* `makeArray` will now return `[]int64{}` instead of `[]time.Duration`

### CODE REVIEW

**Comment by yznima** [comment]
@jkessler93 This is what we have so far

**Comment by yznima** [comment]
@appian/datalayer-ha This is ready for review

File: decode.go [hunk 1]
```diff
@@ -258,50 +258,45 @@ func readData(r *bufio.Reader, order binary.ByteOrder) (kobj *K, err error) {
...
+		case KC:
 			return &K{msgtype, vecattr, string(arr.([]byte))}, nil
-		}
-		if msgtype == KP {
-			arr := arr.([]time.Duration)
+		case KP:
 			var timearr = make([]time.Time, veclen)
-			for i := 0; i < int(veclen); i++ {
-				timearr[i] = qEpoch.Add(arr[i])
+			for i, arr := range arr.([]time.Duration) {
```
Comment (yznima) [inline decode.go]: rename var

File: decode.go [hunk 2]
```diff
@@ -370,7 +361,7 @@ func readData(r *bufio.Reader, order binary.ByteOrder) (kobj *K, err error) {
 		return NewDict(dk, dv), nil
 	case XT:
 		var vecattr Attr
-		err = binary.Read(r, order, &vecattr)
+		err := binary.Read(r, order, &vecattr)
```
Comment (jkessler93) [inline decode.go]: Can we put this into the if like we do in the other places?

File: decode_test.go [hunk 3]
```diff
@@ -7,170 +7,27 @@ import (
...
-			t.Fail()
-		}
-		if !vec[0].Equal(TimestampAsTime) {
-			t.Fail()
-		}
-		if !vec[1].Equal(TimestampAsTime) {
-			t.Fail()
+func TestDecoding(t *testing.T) {
+	for _, tt := range testCases {
+		//fmt.Println(tt.desc)
```
Comment (jkessler93) [inline decode_test.go]: We can remove this

**Review by jkessler93 — APPROVED** [review]
🎆 🍾

File: decode.go [hunk 4]
```diff
@@ -230,26 +226,30 @@ func readData(r *bufio.Reader, order binary.ByteOrder) (kobj *K, err error) {
 		var span time.Duration
 		binary.Read(r, order, &span)
 		return &K{msgtype, NONE, span}, nil
+	case -KD:
+		var days int32
+		binary.Read(r, order, &days)
+		secs := int64(days) * int64(time.Hour*24)
```
Comment (sbenderli) [inline decode.go]: isn't this calculation for hours?

**Review by sbenderli — APPROVED** [review]
LGTM

File: decode.go (hunk 4 above)
Comment (yznima) [reply decode.go]: Great catch. Its actually in `nanos`.

**Comment by yznima** [comment]
@jkessler93 or @sbenderli Addressed all the comments. Ready to merge

**Pull request merged**